projectHashProposal.md

```diff
@@ -0,0 +1,18 @@
+# LSIF: ProjectWithHash
+
+Formulating and processing LSIF for large repositories can be very time consuming. We've discovered that many commits touch only ~20% of projects in a given repository, so there can be lots of unnecessary processing done on unchanged projects that are then ignored at ingestion time. To decrease processing times, I propose adding a new "projectHash" property to the existing [project vertex](https://github.com/Microsoft/language-server-protocol/blob/master/indexFormat/specification.md#the-project-vertex). The new definition for the `project` vertex would be:
+
+```ts
+export interface ProjectWithHash extends Project {
+    /**
+     * An optional hash of the files included in the project, `base64` encoded.
```

Comment:
I wouldn't mention files here since in most cases hash has to be calculated from input source files and references and may also depend on build configuration for example for C++ projects.

We should clarify that projectHash should only be same for 2 projects if their LSIF output is equivalent. I think the explanation below makes that appearent but it should be part of the property comment as well.